models/helper.go

```diff
@@ -13,20 +13,20 @@ import (
 //   a list of filters in the form of concept ids and cohort pairs. The one assumption it makes is that the given `query` object already contains
 //   a basic query on a table or view that have been named or aliased as "observation" (see comments in code). This assumption is
 //   checked at the start.
-func QueryFilterByConceptIdsAndCohortPairsHelper(query *gorm.DB, filterConceptIds []int64, filterCohortPairs []utils.CustomDichotomousVariableDef, omopSchemaName string, resultSchemaName string) *gorm.DB {
+func QueryFilterByConceptIdsAndCohortPairsHelper(query *gorm.DB, filterConceptIds []int64, filterCohortPairs []utils.CustomDichotomousVariableDef, omopDataSource *utils.DbAndSchema, resultSchemaName string) *gorm.DB {
 	// Validate assumption of a table or view aliased as "observation":
-	if query.Statement.Table != "observation" {
-		panic("Error: this QueryFilterByConceptIdsAndCohortPairsHelper is meant for adding extra filters to a query on a table or view with the alias name `observation`")
-	}
+	// if query.Statement.Table != "observation" {
+	// 	panic("Error: this QueryFilterByConceptIdsAndCohortPairsHelper is meant for adding extra filters to a query on a table or view with the alias name `observation`")
+	// }
 
 	// iterate over the filterConceptIds, adding a new INNER JOIN and filters for each, so that the resulting set is the
 	// set of persons that have a non-null value for each and every one of the concepts:
 	for i, filterConceptId := range filterConceptIds {
 		observationTableAlias := fmt.Sprintf("observation_filter_%d", i)
 		log.Printf("Adding extra INNER JOIN with alias %s", observationTableAlias)
-		query = query.Joins("INNER JOIN "+omopSchemaName+".observation_continuous as "+observationTableAlias+" ON "+observationTableAlias+".person_id = observation.person_id"). // assumption: there is a table or view named or aliased as "observation"
-			Where(observationTableAlias+".observation_concept_id = ?", filterConceptId).
-			Where("(" + observationTableAlias + ".value_as_string is not null or " + observationTableAlias + ".value_as_number is not null)") // TODO - improve performance by only filtering on type according to getConceptValueType()
+		query = query.Joins("INNER JOIN "+omopDataSource.Schema+".observation_continuous as "+observationTableAlias+omopDataSource.GetViewDirective()+" ON "+observationTableAlias+".person_id = observation.person_id"). // assumption: there is a table or view named or aliased as "observation"
+																													Where(observationTableAlias+".observation_concept_id = ?", filterConceptId).
```

Comment:
weird spacing

Reply:
I know...the IDE is doing that at every save 😒

Reply:
seems to be magically solved now